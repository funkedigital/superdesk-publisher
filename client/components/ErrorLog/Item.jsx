import React from "react";
import PropTypes from "prop-types";
import moment from "moment";
import classNames from "classnames";

const Item = ({ item }) => {
  const [state, setState] = React.useState({ isExpanded: false, tab: 1 });

  let messageKeys = Object.keys(item.message);

  return (
    <div className="sd-list-item sd-list-item--no-hover sd-shadow--z1">
      <div className="sd-list-item__column sd-list-item__column--grow sd-list-item__column--no-border">
        <div
          className="sd-list-item__row cursorPointer"
          onClick={() => setState({ isExpanded: !state.isExpanded, tab: 1 })}
        >
          <div className="sd-overflow-ellipsis sd-list-item--element-grow">
            <span className="state-label state-in_progress sd-margin-r--1">
              {item.transport}
            </span>
            {item.class}{" "}
            <span className="sd-text__date-time sd-margin-l--1">
              {moment(item.failed_at).format("YYYY-MM-DD, HH:mm:ss")}
            </span>
          </div>
          {state.isExpanded ? (
            <i className="icon-chevron-up-thin sd-margin-l--1"></i>
          ) : (
            <i className="icon-chevron-down-thin sd-margin-l--1"></i>
          )}
        </div>

        {state.isExpanded && (
          <React.Fragment>
            <div
              className="sd-alert sd-alert--hollow sd-alert--alert sd-alert--small"
              style={{ margin: "1em 2em 1em 1em" }}
            >
              {item.error_message}
            </div>

            <div>
              <ul className="nav-tabs">
                <li
                  className={classNames("nav-tabs__tab", {
                    "nav-tabs__tab--active": state.tab === 1
                  })}
                >
                  <button
                    onClick={() => setState({ isExpanded: true, tab: 1 })}
                    className="nav-tabs__link"
                  >
                    <span>Message</span>
                  </button>
                </li>
                <li
                  className={classNames("nav-tabs__tab", {
                    "nav-tabs__tab--active": state.tab === 2
                  })}
                >
                  <button
                    onClick={() => setState({ isExpanded: true, tab: 2 })}
                    className="nav-tabs__link"
                  >
                    <span>Stacktrace</span>
                  </button>
                </li>
                <li
                  className={classNames("nav-tabs__tab", {
                    "nav-tabs__tab--active": state.tab === 3
                  })}
                >
                  <button
                    onClick={() => setState({ isExpanded: true, tab: 3 })}
                    className="nav-tabs__link"
                  >
                    <span>Redeliveries</span>
                  </button>
                </li>
              </ul>
              <div className="nav-tabs__content">
                {state.tab === 1 && (
                  <div className="nav-tabs__pane">
                    {messageKeys.map((key, index) => (
                      <div key={"message" + index + "-" + item.id}>
                        <strong>{key}: </strong>
                        <pre>{item.message[key]}</pre>
                      </div>
                    ))}
                  </div>
                )}
                {state.tab === 2 && (
                  <div className="nav-tabs__pane">
                    <pre>{item.exception_stacktrace}</pre>
                  </div>
                )}
                {state.tab === 3 && (
                  <div className="nav-tabs__pane">
                    <ul>
                      {item.redeliveries.map((r, index) => (
                        <li key={"redelivery" + index + "-" + item.id}>
                          {moment(r).format("YYYY-MM-DD, HH:mm:ss")}
                        </li>
                      ))}
                    </ul>
                  </div>
                )}
              </div>
            </div>
          </React.Fragment>
        )}
      </div>
    </div>
  );
};

Item.propTypes = {
  item: PropTypes.object.isRequired
};

export default Item;
