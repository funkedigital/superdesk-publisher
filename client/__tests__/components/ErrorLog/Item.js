import React from "react";
import Item from "../../../components/ErrorLog/Item";
import { render, fireEvent } from "@testing-library/react";

const item = {
  "id": 5,
  "class": "SWP\\Bundle\\CoreBundle\\MessageHandler\\Message\\ContentPushMessage",
  "failed_at": "2020-03-02T13:07:42+00:00",
  "error_message": "Argument 1 passed to SWP\\Bundle\\CoreBundle\\MessageHandler\\AbstractContentPushHandler::__construct() must implement interface Symfony\\Component\\Messenger\\MessageBusInterface, instance of Symfony\\Bridge\\Monolog\\Logger given, called in \/private\/var\/www\/publisher\/var\/cache\/dev\/ContainerADrp4xz\/getContentPushHandlerService.php on line 17",
  "transport": "async_content_push",
  "redeliveries": [
    "2020-03-02T13:07:34+00:00",
    "2020-03-02T13:07:35+00:00",
    "2020-03-02T13:07:38+00:00",
    "2020-03-02T13:07:42+00:00"
  ],
  "message": {
    "tenant": 1,
    "content": "{\n      \"language\":\"en\",\n      \"slugline\":\"abstract-html-test\",\n      \"body_html\":\"<p>some html body edit 3 raf<\/p>\",\n      \"versioncreated\":\"2016-09-23T13:57:28+0000\",\n      \"firstcreated\":\"2016-05-25T10:23:15+0000\",\n      \"description_text\":\"some abstract text\",\n      \"place\":[\n        {\n          \"country\":\"Australia\",\n          \"world_region\":\"Oceania\",\n          \"state\":\"Australian Capital Territory\",\n          \"qcode\":\"ACT\",\n          \"name\":\"ACT\",\n          \"group\":\"Australia\"\n        }\n      ],\n      \"version\":\"2\",\n      \"byline\":\"ADmin\",\n      \"keywords\":[\n        \"keyword1\",\n        \"keyword2\"\n      ],\n      \"guid\":\"urn:newsml:localhost:2016-09-23T13:56:39.404843:56465de4-0d5c-495a-8e36-3b396def3cf0\",\n      \"priority\":6,\n      \"subject\":[\n        {\n          \"name\":\"lawyer\",\n          \"code\":\"02002001\"\n        }\n      ],\n      \"urgency\":3,\n      \"type\":\"text\",\n      \"headline\":\"testing correction\",\n      \"service\":[\n        {\n          \"name\":\"Australian General News\",\n          \"code\":\"a\"\n        }\n      ],\n      \"description_html\":\"<p><b><u>some abstract text<\/u><\/b><\/p>\",\n      \"located\":\"Warsaw\",\n      \"pubstatus\":\"usable\"\n    }"
  },
  "exception_stacktrace": "#0 \/private\/var\/www\/publisher\/var\/cache\/dev\/ContainerADrp4xz\/getContentPushHandlerService.php(17): SWP\\Bundle\\CoreBundle\\MessageHandler\\AbstractContentPushHandler->__construct(Object(Symfony\\Bridge\\Monolog\\Logger), Object(SWP\\Bundle\\CoreBundle\\Repository\\PackageRepository), Object(Symfony\\Component\\HttpKernel\\Debug\\TraceableEventDispatcher), Object(SWP\\Component\\Bridge\\Transformer\\JsonToPackageTransformer), Object(Pixers\\DoctrineProfilerBundle\\ORM\\EntityManager), Object(SWP\\Bundle\\CoreBundle\\Context\\CachedTenantContext), Object(Sentry\\State\\Hub), Object(SWP\\Bundle\\CoreBundle\\Hydrator\\PackageHydrator), Object(SWP\\Bundle\\CoreBundle\\Hydrator\\PackageHydrator))\n#1 \/private\/var\/www\/publisher\/var\/cache\/dev\/ContainerADrp4xz\/srcApp_KernelDevDebugContainer.php(1446): require('\/private\/var\/ww...')\n#2 \/private\/var\/www\/publisher\/var\/cache\/dev\/ContainerADrp4xz\/get_Messenger_HandlerDescriptor_IktyVtDService.php(11): ContainerADrp4xz\\srcApp_KernelDevDebugContainer->load('getContentPushH...')\n#3 \/private\/var\/www\/publisher\/var\/cache\/dev\/ContainerADrp4xz\/srcApp_KernelDevDebugContainer.php(1446): require('\/private\/var\/ww...')\n#4 \/private\/var\/www\/publisher\/var\/cache\/dev\/ContainerADrp4xz\/getCommand_Bus_Middleware_HandleMessageService.php(21): ContainerADrp4xz\\srcApp_KernelDevDebugContainer->load('get_Messenger_H...')\n#5 \/private\/var\/www\/publisher\/vendor\/symfony\/messenger\/Handler\/HandlersLocator.php(43): ContainerADrp4xz\\srcApp_KernelDevDebugContainer->{closure}()\n#6 \/private\/var\/www\/publisher\/vendor\/symfony\/messenger\/Middleware\/HandleMessageMiddleware.php(56): Symfony\\Component\\Messenger\\Handler\\HandlersLocator->getHandlers(Object(Symfony\\Component\\Messenger\\Envelope))\n#7 \/private\/var\/www\/publisher\/vendor\/symfony\/messenger\/Middleware\/SendMessageMiddleware.php(79): Symfony\\Component\\Messenger\\Middleware\\HandleMessageMiddleware->handle(Object(Symfony\\Component\\Messenger\\Envelope), Object(Symfony\\Component\\Messenger\\Middleware\\TraceableStack))\n#8 \/private\/var\/www\/publisher\/vendor\/symfony\/messenger\/Middleware\/ValidationMiddleware.php(48): Symfony\\Component\\Messenger\\Middleware\\SendMessageMiddleware->handle(Object(Symfony\\Component\\Messenger\\Envelope), Object(Symfony\\Component\\Messenger\\Middleware\\TraceableStack))\n#9 \/private\/var\/www\/publisher\/vendor\/symfony\/messenger\/Middleware\/FailedMessageProcessingMiddleware.php(34): Symfony\\Component\\Messenger\\Middleware\\ValidationMiddleware->handle(Object(Symfony\\Component\\Messenger\\Envelope), Object(Symfony\\Component\\Messenger\\Middleware\\TraceableStack))\n#10 \/private\/var\/www\/publisher\/vendor\/symfony\/messenger\/Middleware\/DispatchAfterCurrentBusMiddleware.php(67): Symfony\\Component\\Messenger\\Middleware\\FailedMessageProcessingMiddleware->handle(Object(Symfony\\Component\\Messenger\\Envelope), Object(Symfony\\Component\\Messenger\\Middleware\\TraceableStack))\n#11 \/private\/var\/www\/publisher\/vendor\/symfony\/messenger\/Middleware\/RejectRedeliveredMessageMiddleware.php(42): Symfony\\Component\\Messenger\\Middleware\\DispatchAfterCurrentBusMiddleware->handle(Object(Symfony\\Component\\Messenger\\Envelope), Object(Symfony\\Component\\Messenger\\Middleware\\TraceableStack))\n#12 \/private\/var\/www\/publisher\/vendor\/symfony\/messenger\/Middleware\/AddBusNameStampMiddleware.php(37): Symfony\\Component\\Messenger\\Middleware\\RejectRedeliveredMessageMiddleware->handle(Object(Symfony\\Component\\Messenger\\Envelope), Object(Symfony\\Component\\Messenger\\Middleware\\TraceableStack))\n#13 \/private\/var\/www\/publisher\/vendor\/symfony\/messenger\/Middleware\/TraceableMiddleware.php(43): Symfony\\Component\\Messenger\\Middleware\\AddBusNameStampMiddleware->handle(Object(Symfony\\Component\\Messenger\\Envelope), Object(Symfony\\Component\\Messenger\\Middleware\\TraceableStack))\n#14 \/private\/var\/www\/publisher\/vendor\/symfony\/messenger\/MessageBus.php(80): Symfony\\Component\\Messenger\\Middleware\\TraceableMiddleware->handle(Object(Symfony\\Component\\Messenger\\Envelope), Object(Symfony\\Component\\Messenger\\Middleware\\TraceableStack))\n#15 \/private\/var\/www\/publisher\/vendor\/symfony\/messenger\/TraceableMessageBus.php(41): Symfony\\Component\\Messenger\\MessageBus->dispatch(Object(Symfony\\Component\\Messenger\\Envelope), Array)\n#16 \/private\/var\/www\/publisher\/vendor\/symfony\/messenger\/RoutableMessageBus.php(54): Symfony\\Component\\Messenger\\TraceableMessageBus->dispatch(Object(Symfony\\Component\\Messenger\\Envelope), Array)\n#17 \/private\/var\/www\/publisher\/vendor\/symfony\/messenger\/Worker.php(117): Symfony\\Component\\Messenger\\RoutableMessageBus->dispatch(Object(Symfony\\Component\\Messenger\\Envelope))\n#18 \/private\/var\/www\/publisher\/vendor\/symfony\/messenger\/Worker.php(81): Symfony\\Component\\Messenger\\Worker->handleMessage(Object(Symfony\\Component\\Messenger\\Envelope), Object(Symfony\\Component\\Messenger\\Transport\\AmqpExt\\AmqpTransport), 'async_content_p...')\n#19 \/private\/var\/www\/publisher\/vendor\/symfony\/messenger\/Command\/ConsumeMessagesCommand.php(202): Symfony\\Component\\Messenger\\Worker->run(Array)\n#20 \/private\/var\/www\/publisher\/vendor\/symfony\/console\/Command\/Command.php(255): Symfony\\Component\\Messenger\\Command\\ConsumeMessagesCommand->execute(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#21 \/private\/var\/www\/publisher\/vendor\/symfony\/console\/Application.php(1030): Symfony\\Component\\Console\\Command\\Command->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#22 \/private\/var\/www\/publisher\/vendor\/symfony\/framework-bundle\/Console\/Application.php(97): Symfony\\Component\\Console\\Application->doRunCommand(Object(Symfony\\Component\\Messenger\\Command\\ConsumeMessagesCommand), Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#23 \/private\/var\/www\/publisher\/vendor\/symfony\/console\/Application.php(272): Symfony\\Bundle\\FrameworkBundle\\Console\\Application->doRunCommand(Object(Symfony\\Component\\Messenger\\Command\\ConsumeMessagesCommand), Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#24 \/private\/var\/www\/publisher\/vendor\/symfony\/framework-bundle\/Console\/Application.php(83): Symfony\\Component\\Console\\Application->doRun(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#25 \/private\/var\/www\/publisher\/vendor\/symfony\/console\/Application.php(148): Symfony\\Bundle\\FrameworkBundle\\Console\\Application->doRun(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#26 \/private\/var\/www\/publisher\/bin\/console(40): Symfony\\Component\\Console\\Application->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput))\n#27 {main}"
};

describe("Analytics/ArticleItem", () => {
  it("renders correctly", () => {
    const { container } = render(<Item item={item} />);

    const expandButton = container.querySelector(".cursorPointer");
    fireEvent.click(expandButton);

    expect(container.firstChild).toMatchSnapshot();
  });
});
