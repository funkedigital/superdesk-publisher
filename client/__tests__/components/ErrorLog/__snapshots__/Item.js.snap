// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analytics/ArticleItem renders correctly 1`] = `
<div
  class="sd-list-item sd-list-item--no-hover sd-shadow--z1"
>
  <div
    class="sd-list-item__column sd-list-item__column--grow sd-list-item__column--no-border"
  >
    <div
      class="sd-list-item__row cursorPointer"
    >
      <div
        class="sd-overflow-ellipsis sd-list-item--element-grow"
      >
        <span
          class="state-label state-in_progress sd-margin-r--1"
        >
          async_content_push
        </span>
        SWP\\Bundle\\CoreBundle\\MessageHandler\\Message\\ContentPushMessage
         
        <span
          class="sd-text__date-time sd-margin-l--1"
        >
          2020-03-02, 14:07:42
        </span>
      </div>
      <i
        class="icon-chevron-up-thin sd-margin-l--1"
      />
    </div>
    <div
      class="sd-alert sd-alert--hollow sd-alert--alert sd-alert--small"
      style="margin: 1em 2em 1em 1em;"
    >
      Argument 1 passed to SWP\\Bundle\\CoreBundle\\MessageHandler\\AbstractContentPushHandler::__construct() must implement interface Symfony\\Component\\Messenger\\MessageBusInterface, instance of Symfony\\Bridge\\Monolog\\Logger given, called in /private/var/www/publisher/var/cache/dev/ContainerADrp4xz/getContentPushHandlerService.php on line 17
    </div>
    <div>
      <ul
        class="nav-tabs"
      >
        <li
          class="nav-tabs__tab nav-tabs__tab--active"
        >
          <button
            class="nav-tabs__link"
          >
            <span>
              Message
            </span>
          </button>
        </li>
        <li
          class="nav-tabs__tab"
        >
          <button
            class="nav-tabs__link"
          >
            <span>
              Stacktrace
            </span>
          </button>
        </li>
        <li
          class="nav-tabs__tab"
        >
          <button
            class="nav-tabs__link"
          >
            <span>
              Redeliveries
            </span>
          </button>
        </li>
      </ul>
      <div
        class="nav-tabs__content"
      >
        <div
          class="nav-tabs__pane"
        >
          <div>
            <strong>
              tenant
              : 
            </strong>
            <pre>
              1
            </pre>
          </div>
          <div>
            <strong>
              content
              : 
            </strong>
            <pre>
              {
      "language":"en",
      "slugline":"abstract-html-test",
      "body_html":"&lt;p&gt;some html body edit 3 raf&lt;/p&gt;",
      "versioncreated":"2016-09-23T13:57:28+0000",
      "firstcreated":"2016-05-25T10:23:15+0000",
      "description_text":"some abstract text",
      "place":[
        {
          "country":"Australia",
          "world_region":"Oceania",
          "state":"Australian Capital Territory",
          "qcode":"ACT",
          "name":"ACT",
          "group":"Australia"
        }
      ],
      "version":"2",
      "byline":"ADmin",
      "keywords":[
        "keyword1",
        "keyword2"
      ],
      "guid":"urn:newsml:localhost:2016-09-23T13:56:39.404843:56465de4-0d5c-495a-8e36-3b396def3cf0",
      "priority":6,
      "subject":[
        {
          "name":"lawyer",
          "code":"02002001"
        }
      ],
      "urgency":3,
      "type":"text",
      "headline":"testing correction",
      "service":[
        {
          "name":"Australian General News",
          "code":"a"
        }
      ],
      "description_html":"&lt;p&gt;&lt;b&gt;&lt;u&gt;some abstract text&lt;/u&gt;&lt;/b&gt;&lt;/p&gt;",
      "located":"Warsaw",
      "pubstatus":"usable"
    }
            </pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
